WIP: Test ci UI smoke tests

ybott-qinetic opened the pull request.

blammit committed:
Tests: support automated UI testing

Add C++ and QML API to support simple automated UI testing. The main
idea is to click through the UI and do a simple image comparison to
verify whether screens have changed.

See tests/ui/README.md for more information.

A basic set of tests have been implemented for mock mode, in
tests/ui/smoke/mock-maximal.

Contributes to #984

ybott committed:
WIP: Test CI based UI test

ybott committed:
WIP: Test CI based UI test

chriadam commented:
You can change the base branch from `main` to Bea's branch, to reduce the diff / delta shown in review

blammit committed:
Tests: support automated UI testing

Add C++ and QML API to support simple automated UI testing. The main
idea is to click through the UI and do a simple image comparison to
verify whether screens have changed.

See tests/ui/README.md for more information.

A basic set of tests have been implemented for mock mode, in
tests/ui/smoke/mock-maximal.

Also, avoid loading the genset and alternator pages asynchronously, as
that causes uncertainty as to when the page content is loaded and
captures are possible for testing.

Contributes to #984
